website/docs/r/vm_internal_disk.html.markdown

```diff
@@ -0,0 +1,107 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vm_internal_disk"
+sidebar_current: "docs-vcd-vm-internal-disk"
+description: |-
+  Provides a vCloud Director VM internal disk resource. This can be used to create and delete VM internal disks.
+---
+
+# vcd\_vm\_internal\_disk
+
+Provides a vCloud Director VM internal disk resource. This can be used to create and delete VM internal disks.
+
+Supported in provider *v2.6+*
+
+## Example Usage
+
+```
+resource "vcd_vm_internal_disk" "disk1" {
+  vapp_name     = "my-vapp"
+  vm_name     = "my-vm1"
+  bus_type = "sata"
+  size_in_mb = "13333"
+  bus_number = 0
+  unit_number = 1
+  storage_profile = "Development"
+  allow_vm_reboot = true
+  depends_on   = ["vcd_vapp_vm.web1"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level
+* `vapp_name` - (Required) The vApp this VM internal disk belongs to.
+* `vm_name` - (Required) VM in vApp in which internal disk created.
+* `allow_vm_reboot` - (Optional) Powers off VM when changing any attribute of an IDE disk or unit/bus number of other disk types, after the change is complete VM is powered back on. Without this setting enabled, such changes on a powered on VM would fail.
+* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`. 
+* `size_in_mb` - (Required) The size of the disk in MB. 
+* `bus_number` - (Required) The number of the SCSI or IDE controller itself.
+* `unit_number` - (Required) The device number on the SCSI or IDE controller of the disk.
+* `iops` - (Optional) Specifies the IOPS for the disk. Default - 0.
+* `storage_profile` - (Optional) Storage profile which overrides the VM default one.
```

Comment:
I'm missing `thin_provisioned`. At least it will be read:

```
-/+ resource "vcd_vm_internal_disk" "tf-myInternalDisk" {
        bus_number       = 0
        bus_type         = "paravirtual"
      ~ id               = "2000" -> (known after apply)
      ~ iops             = 0 -> (known after apply)
      - org              = "XXX" -> null # forces replacement
        size_in_mb       = 40960
      ~ storage_profile  = "XXX" -> (known after apply)
      ~ thin_provisioned = true -> (known after apply)
        unit_number      = 0
        vapp_name        = "test1"
      - vdc              = "XXX" -> null # forces replacement
        vm_name          = "test1"
    }
```

Reply:
It is in section a little below:

## Attribute reference
* `thin_provisioned` - Specifies whether the disk storage is pre-allocated or allocated on demand.

Reply:
I was confused because for  `override_template_disk` the parameter
`thin_provisioned` is configurable. But this is VCD feature.
-> OK

Reply:
@buec011 If you're happy with the outcome, please press "Resolve conversation" below. The comments on this PR are starting to run away :)